### DISCUSSION TITLE
[Config Support]: Using json Schema cannot get it working

### DISCUSSION THREAD

**Discussion by plug-it-in** [blakeblackshear/frigate#13595] [Config Support]
### Describe the problem you are having

So this is about these instructions in your help document

VSCode Configuration Schema
VSCode supports JSON schemas for automatically validating configuration files. You can enable this feature by adding # yaml-language-server: $schema=http://frigate_host:5000/api/config/schema.json to the beginning of the configuration file. Replace frigate_host with the IP address or hostname of your Frigate server. If you're using both VSCode and Frigate as an add-on, you should use ccab4aaf-frigate instead. Make sure to expose the internal unauthenticated port 5000 when accessing the config from VSCode on another machine.

Being a newbie using both VSCode and Frigate as HA Addons I cannot get this to work because the instructions are unclear

When you say use ccab4aaf-frigate instead. Instead of what? The IP address ? Ive tried 
$schema=http://ccab4aaf-frigate/api/config/schema.json it fails

When you say expose internal unathenticated port 5000 can you please include how to do this I have no idea what this means and where to do it. If its in the Frigate Configuration, can you please say this and which in the list I should open. For example is this the unauthenticated Web port should I set that to 5000

Getting this all working for a newbie is very hard and frustrating.

A simple example URL for my situation would save a huge amount of time 

### Version

0.14,1

### Frigate config file

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Relevant go2rtc log output

```shell
NA
```

### Frigate stats

```json
NA
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

CPU (no coral)

### Screenshots of the Frigate UI's System metrics pages

NA

**Comment by NickM-27** [comment]
it's easiest if you just use the built in config editor which has yaml validation / linting built in. 

You enable port 5000 in the addon settings under extra configuration if you are running as an addon

**Reply by plug-it-in** [reply]
See  its this kind of info that really doesnt help me. Can you please be more specific.  What addon are you talking about. VSCode or Frigate?  And where is this illusive extra configuration?

Im trying to get the Frigate HA Integration working and I have come up against the same problem as well it states 

Frigate
URL you use to access Frigate (ie. http://frigate:5000/)

If you are using HassOS with the addon, the URL should be http://ccab4aaf-frigate:5000/
Home Assistant needs access to port 5000 (api) and 1935 (rtmp) for all features.
The integration will setup sensors, cameras, and media browser functionality.

Here is another unclear statement that drives me nuts
**Home Assistant needs access to port 5000 (api) and 1935 (rtmp) for all features.**

How and where 

I do not understand whilst your going to all this effort to write so much good help why not add this final bit of info that would certainly stop me bugging you.

**Reply by plug-it-in** [reply]
Oh forgot to mention I cannot use the inbuilt editor all the time as that relies on Frigate starting. Also the update and restart from this doesnt work.

**Reply by NickM-27** [reply]
I don't use HA OS so I can't really get specific, HA also changes their UI fairly often and we don't want to have to keep updating our docs when something with their UI changes. You are looking for extra configuration in the frigate addon. It is there in the addon settings.

> Here is another unclear statement that drives me nuts
> Home Assistant needs access to port 5000 (api) and 1935 (rtmp) for all features.

where do you see that? that is certainly not in frigate's documentation. 

Since you have frigate installed as an addon, all you have to do to setup the integration is enter the url that corresponds to the type of addon that you installed. They are all listed at https://docs.frigate.video/integrations/home-assistant#hassos-addon

**Reply by NickM-27** [reply]
if you only ever use the built in editor it won't let you save an invalid config. If the save and restart button is not working then that would be a bug report, we have not had anyone else report this issue

**Comment by gavdotcom** [comment]
"When you say use ccab4aaf-frigate instead. Instead of what?"

The answer is instead of the IP address.  Docker instances (what HA uses internally) can and should refer to each other by name. So you are going to use one of the following, depending on what Frigate HA app you installed, which dictates which Docker name it has:

```
# yaml-language-server: $schema=http://ccab4aaf-frigate:5000/api/config/schema.json
# yaml-language-server: $schema=http://ccab4aaf-frigate-fa:5000/api/config/schema.json
# yaml-language-server: $schema=http://ccab4aaf-frigate-beta:5000/api/config/schema.json
# yaml-language-server: $schema=http://ccab4aaf-frigate-beta-fa:5000/api/config/schema.json
```

More info on those Docker container names here:
[https://docs.frigate.video/configuration/](https://docs.frigate.video/configuration/)

The port 5000 is settable under Home Assistant's Settings > Apps > Frigate > Configuration... then down below "Show disabled ports"

This port description says "Unauthenticated Web interface (not required for Home Assistant Ingress)". 

Enabling a port just maps it to the external LAN IP (on your local network) and doing so would give your entire LAN to your Frigate setup with no username and password. Far from ideal. You don't need to do this because port 5000 is accessible by Home Assistant and VS Code already, which is how you can show Frigate in the Home Assistant sidebar, and click on it to get in without being prompted for a username and password. Frigate describes this port as "Internal unauthenticated UI and API access. Access to this port should be limited. Intended to be used within the docker network for services that integrate with Frigate."

In VS Code you may need to go to its Settings > Extensions > JSON  and enable downloads and whitelist Frigate or add a wildcard in order to be able to download the schema.
